appcues/src/main/java/com/appcues/RenderContextManager.kt

```diff
@@ -1,25 +1,80 @@
 package com.appcues
 
+import com.appcues.analytics.ExperienceLifecycleTracker
+import com.appcues.data.model.Experience
+import com.appcues.data.model.ExperienceTrigger
+import com.appcues.data.model.ExperienceTrigger.Qualification
 import com.appcues.data.model.RenderContext
 import com.appcues.data.model.RenderContext.Embed
+import com.appcues.statemachine.StateMachine
+import org.koin.core.component.KoinScopeComponent
+import org.koin.core.component.get
+import org.koin.core.component.inject
+import org.koin.core.scope.Scope
 import java.lang.ref.WeakReference
 
-internal class RenderContextManager {
+internal class RenderContextManager(override val scope: Scope) : KoinScopeComponent {
```

Comment:
initially had RenderContextManager only for storing the AppcuesFrameView but I think its better to have it responsible for everything related to renderContext, instead of having half of the work here and the other half on ExperienceRenderer